packages/mdl-base-component/adapter.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * A basic component adapter.
+ */
+export const MDLBaseAdapter = {
+  getBoundingClientRect(id) {
+    return this.elements_[id].getBoundingClientRect();
+  },
+
+  setStyles(id, styles) {
+    Object.assign(this.elements_[id].style, styles);
+  },
+
+  forceLayout(id) {
+    return this.elements_[id].offsetWidth;
+  },
+
+  addClass(id, cls) {
+    this.elements_[id].classList.add(cls);
+  },
+
+  removeClass(id, cls) {
+    this.elements_[id].classList.remove(cls);
+  },
+
+  get(...symbolPath) {
```

Comment:
The base adapter here would essentially be used for DOM-based renderers. I assume for other cases, they'd use a completely different class translating the calls to the underlying objects. The radio button doesn't really need this functionality, but I can't think of a straightforward way to avoid significant code duplication for the ripple, which needs to perform operations on a bunch of different elements.